Data/Table/Row.idr

```diff
@@ -11,6 +11,26 @@ import public Data.Table.Row.HasRows
 
 %default total
 
+export
+ByProof : forall table.
+     (hasRows : table `HasRows` n)
+  -> (prf : n = m) -> (table `HasRows` m)
+ByProof {table} hasRows prf = replace {p = \k => table `HasRows` k} prf hasRows
+
+public export 0
```

Comment:
We can probably allow this function at runtime, as `HasRows tbl n` is `n` at runtime